test/cat.jl

```diff
@@ -155,14 +175,169 @@ end
 end
 
 @testset "vcat >2 args" begin
-    empty_dfs = [DataFrame(), DataFrame(), DataFrame()]
+    empty_dfs = [DataFrame(), DataFrame(), DataFrame()]         
+    df1 = DataFrame(A = 1:3, B = 4:6, C = 7:9)
     @test vcat(empty_dfs...) == reduce(vcat, empty_dfs) == DataFrame()
 
     df = DataFrame(x = trues(1), y = falses(1))
     dfs = [df, df, df]
     @test vcat(dfs...) ==reduce(vcat, dfs) == DataFrame(x = trues(3), y = falses(3))
 end
 
+    @testset "vcat out of order" begin
+        df1 = DataFrame(A = 1:3, B = 4:6, C = 7:9)
+        df2 = DataFrame(colwise(x->2x, df1), reverse(names(df1)))
+        @test vcat(df1, df2) == DataFrame(A = [1, 2, 3, 14, 16, 18],
+                                          B = [4, 5, 6, 8, 10, 12],
+                                          C = [7, 8, 9, 2, 4, 6])
+        # test with keyword argument for `columns`
+        @test vcat(df1, df2, columns = :equal) == DataFrame(A = [1, 2, 3, 14, 16, 18],
+                                                           B = [4, 5, 6, 8, 10, 12],
+                                                           C = [7, 8, 9, 2, 4, 6])
+        @test vcat(df1, df1, df2) == DataFrame(A = [1, 2, 3, 1, 2, 3, 14, 16, 18],
+                                               B = [4, 5, 6, 4, 5, 6, 8, 10, 12],
+                                               C = [7, 8, 9, 7, 8, 9, 2, 4, 6])
+        @test vcat(df1, df2, df2) == DataFrame(A = [1, 2, 3, 14, 16, 18, 14, 16, 18],
+                                               B = [4, 5, 6, 8, 10, 12, 8, 10, 12],
+                                               C = [7, 8, 9, 2, 4, 6, 2, 4, 6])
+        @test vcat(df2, df1, df2) == DataFrame(C = [2, 4, 6, 7, 8, 9, 2, 4, 6],
+                                               B = [8, 10, 12, 4, 5, 6, 8, 10, 12],
+                                               A = [14, 16, 18, 1, 2, 3, 14, 16, 18]) 
+        @test size(vcat(df1, df1, df1, df2, df2, df2)) == (18, 3)
+        df3 = df1[[1, 3, 2]]
+        res = vcat(df1, df1, df1, df2, df2, df2, df3, df3, df3, df3)
+        @test size(res) == (30, 3)
+        @test res[1:3,:] == df1
+        @test res[4:6,:] == df1
+        @test res[7:9,:] == df1
+        @test res[10:12,:] == df2[names(res)]
+        @test res[13:15,:] == df2[names(res)]
+        @test res[16:18,:] == df2[names(res)]
+        @test res[19:21,:] == df3[names(res)]
+        @test res[22:24,:] == df3[names(res)]
+        @test res[25:27,:] == df3[names(res)]
+        df1 = DataFrame(A = 1, B = 2)
+        df2 = DataFrame(B = 12, A = 11)
+        df3 = DataFrame(A = [1, 11], B = [2, 12])
+        @test [df1; df2] == df3
+    end
+
+    @testset "vcat with :union" begin
+        df1 = DataFrame(A = 1:3, B = 4:6)
+        df2 = DataFrame(A = 7:9)
+        df3 = DataFrame(B = 4:6, A = 1:3)
+
+        @test vcat(df1, df2; columns = :union) ≅ DataFrame(A =Int[1, 2, 3, 7, 8, 9], 
```

Comment:
No need to explicitly write types. Also add space before `=`, and a line break before `DataFrame` to avoid going beyond 92 chars.

Reply:
changed.